app/src/main/java/nextstep/jwp/controller/LoginController.java

```diff
@@ -4,43 +4,53 @@
 import nextstep.jwp.db.InMemoryUserRepository;
 import nextstep.jwp.exception.DBNotFoundException;
 import nextstep.jwp.exception.PasswordMismatchException;
-import nextstep.jwp.framework.http.HttpBody;
-import nextstep.jwp.framework.http.HttpRequest;
-import nextstep.jwp.framework.http.HttpResponse;
-import nextstep.jwp.framework.http.HttpStatus;
+import nextstep.jwp.framework.http.common.HttpStatus;
+import nextstep.jwp.framework.http.request.HttpRequest;
+import nextstep.jwp.framework.http.response.HttpResponse;
+import nextstep.jwp.framework.http.session.HttpSession;
+import nextstep.jwp.framework.http.session.HttpSessions;
 import nextstep.jwp.model.User;
 
 public class LoginController extends AbstractController {
 
     @Override
     protected void doGet(HttpRequest request, HttpResponse response) {
+        checkAlreadyLogin(request);
         response.create(request.getRequestLine(), request.getHeaders(), request.getBody(), HttpStatus.OK);
     }
 
     @Override
     protected void doPost(HttpRequest request, HttpResponse response) {
-        final HttpStatus status = login(request.getBody());
+        final HttpStatus status = login(request);
         response.create(request.getRequestLine(), request.getHeaders(), request.getBody(), status);
     }
 
-    private HttpStatus login(final HttpBody body) {
+    private HttpStatus login(final HttpRequest request) {
         try {
-            checkAccount(body);
-            return HttpStatus.FOUND;
+            return checkAccount(request);
         } catch (DBNotFoundException | PasswordMismatchException ignore) {
             return HttpStatus.UNAUTHORIZED;
         }
     }
 
-    private void checkAccount(final HttpBody body) {
-        final Map<String, String> queryParams = body.getQueryParams();
+    private void checkAlreadyLogin(final HttpRequest request) {
+        if (HttpSessions.isLoggedIn(request.sessionId())) {
+            request.changeIndexPage();
+        }
+    }
+
+    private HttpStatus checkAccount(final HttpRequest request) {
+        final Map<String, String> queryParams = request.getQueryParams();
         final String account = queryParams.get("account");
         final String password = queryParams.get("password");
         final User user = InMemoryUserRepository.findByAccount(account).orElseThrow(DBNotFoundException::new);
 
-        logger.debug(account + "님이 접속했습니다.");
         if (user.checkPassword(password)) {
-            return;
+            logger.debug(account + "님이 접속했습니다.");
```

Comment:
SonarLint에서 다른 방식으로 문자열을 합치라고 얘기해주네요!
이 부분은 저도 이것저것 하면서 알게 된 부분인데요, 해당 메서드가 실행되어 로깅을 수행할 때마다 스트링을 합치는 연산이 실행되어 성능에 조금이나마 손실이 발생하기 때문이라고 합니다. (SonarLint에서 알려줬어요 ㅋㅋㅋ)
그래서 Slf4j에서 지원해주는 방식으로 로그를 남기는 것이 더 효율적이니까 다음과 같이 바꿔 볼 수 있겠네요!
다른 로깅 부분도 확인해주세요 😁
```suggestion
            logger.debug("{}님이 접속했습니다.", account);
```

Reply:
오호 그런 이유가 있었군요!